--- a/arch/arm64/boot/dts/qcom/ipq8074.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq8074.dtsi
@@ -108,6 +108,39 @@
 			reg = <0x0 0x4ac00000 0x0 0x400000>;
 			no-map;
 		};
+
+		tzapp_region: tzapp@4b000000 {
+			reg = <0x0 0x4b000000 0x0 0x400000>;
+			no-map;
+		};
+
+		q6_region: q6@4b400000 {
+			reg = <0x0 0x4b400000 0x0 0x6000000>;
+			no-map;
+		};
+
+		q6_etr_region: q6_etr@51400000 {
+			reg = <0x0 0x51400000 0x0 0x100000>;
+			no-map;
+		};
+
+		m3_dump_region: m3_dump@51500000 {
+			reg = <0x0 0x51500000 0x0 0x100000>;
+			no-map;
+		};
+
+		ramoops_region: ramoops@51600000 {
+			compatible = "ramoops";
+			reg = <0x0 0x51600000 0x0 0x100000>;
+			no-map;
+			record-size = <0x4000>;
+			console-size = <0x4000>;
+		};
+
+		nss_region: nss@40000000 {
+			reg = <0x0 0x40000000 0x0 0x1000000>;
+			no-map;
+		};
 	};
 
 	firmware {
